src/components/pages/admin/contacts/RoleEditor.vue

```diff
@@ -0,0 +1,167 @@
+            <AppRadioGroup
+              v-model="roleHasStartDate"
+              name="roleStartDate"
+              :options="[
+                [false, inputT('starts.opts.now')],
+                [true, inputT('starts.opts.schedule')],
+              ]"
+              required
+            />
+            <div v-if="roleHasStartDate" class="flex gap-2">
+              <div>
+                <AppInput v-model="newRole.startDate" type="date" required />
+              </div>
+              <div>
+                <AppInput v-model="newRole.startTime" type="time" required />
+              </div>
+            </div>
+          </div>
+
+          <div class="my-2 py-1">
+            <AppRadioGroup
+              v-model="roleHasEndDate"
+              name="roleEndDate"
+              :label="inputT('expires.label')"
+              :options="[
+                [false, inputT('expires.opts.never')],
+                [true, inputT('expires.opts.schedule')],
+              ]"
+              required
+            />
+            <div v-if="roleHasEndDate" class="flex gap-2">
+              <div>
+                <AppInput v-model="newRole.endDate" type="date" required />
+              </div>
+              <div>
+                <AppInput v-model="newRole.endTime" type="time" required />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div class="my-3 flex py-3">
+          <AppButton type="submit" variant="primary" :loading="loading">{{
+            t('contacts.data.roles.add')
+          }}</AppButton>
+          <AppButton variant="text" class="ml-2" @click="formVisible = false">{{
+            t('form.cancel')
+          }}</AppButton>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import AppLabel from '../../../../components/forms/AppLabel.vue';
+import AppInput from '../../../../components/forms/AppInput.vue';
+import AppButton from '../../../../components/forms/AppButton.vue';
+import AppSelect from '../../../../components/forms/AppSelect.vue';
+import AppRadioGroup from '../../../../components/forms/AppRadioGroup.vue';
+import RoleEditorItem from '../../../../components/pages/admin/contacts/RoleEditorItem.vue';
+import { GetMemberData } from '../../../../utils/api/api.interface';
+import { updateRole } from '../../../../utils/api/member';
+import { onBeforeMount, ref, reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+const emit = defineEmits(['update']);
+const props = defineProps<{
+  contact: GetMemberData;
+}>();
+
+const { t } = useI18n();
+const inputT = (key: string) => t('contacts.data.roles.' + key);
+
+const loading = ref(false);
+const formVisible = ref(false);
+const roleHasStartDate = ref(false);
+const roleHasEndDate = ref(false);
+const newRole = reactive({
+  role: '',
+  startDate: '',
+  startTime: '',
+  endDate: '',
+  endTime: '',
+});
+
+// FIXME: hardcoded
+const roleOptions = [
```

Comment:
These need to be internationalised properly, you can use `t('common.role.<id>')` for that